lib/tasks/translation.rake

```diff
@@ -87,21 +88,16 @@ namespace :translation do
   desc "Add missing translations"
   task(:add_missing, [:locale] => [:environment]) do |t, args|
     locale = args[:locale]
-    require 'i18n/tasks/cli'
 
-    i18n_args = ["add-missing", "--nil-value"]
-    i18n_args << ["-l", locale] if locale
-
-    I18n::Tasks::CLI.start(i18n_args.flatten)
+    I18n::Tasks::CLI.start(TranslationHelper.new(["add-missing", "--nil-value"], locale).with_optional_locale)
     RailsTranslationManager::Cleaner.new(Rails.root.join("config", "locales")).clean
   end
 
   desc "Normalize translations"
   task(:normalize, [:locale] => [:environment]) do |t, args|
     locale = args[:locale]
-    require 'i18n/tasks/cli'
 
-    locale ? I18n::Tasks::CLI.start(["normalize", "-l", locale]) : I18n::Tasks::CLI.start(["normalize"])
+    I18n::Tasks::CLI.start(TranslationHelper.new(["normalize"], locale).with_optional_locale)
```

Comment:
Does this work? It looks like `I18n::Tasks::CLI.start` takes an array, but we're passing it a `TranslationHelper` object.

Don't we need to define a `to_a` on `TranslationHelper` to return the `@task_options` array?

Reply:
@ChrisBAshton Here, on `TranslationHelper` instance we call `with_optional_locale` method which returns an array.

Reply:
Thanks for explaining - I forgot about the implicit return!